src/readMSH.cpp

```diff
@@ -44,25 +37,26 @@ npe_arg(filename, std::string)
 npe_default_arg(dtypef, npe::dtype, "float")
 
 npe_begin_code()
+  Eigen::MatrixXd v;
+  Eigen::MatrixXi t;
+
+  //NOTE: readMSH only support vertices as matrix of doubles. We instead cast the output accordingly.
+  bool ret = igl::readMSH(filename, v, t);
+  if (!ret) {
+    throw std::invalid_argument("File '" + filename + "' not found.");
+  }
+  
+  EigenDenseInt t_row_maj = t.template cast<typename EigenDenseInt::Scalar>();
+
   if (dtypef.type() == npe::type_f32) {
-    EigenDenseF32 v;
-    EigenDenseInt t;
-    bool ret = igl::readMSH(filename, v, t);
-    if (!ret) {
-      throw std::invalid_argument("File '" + filename + "' not found.");
-    }
-    return std::make_tuple(npe::move(v), npe::move(t));
+    EigenDenseF32 v_row_maj = v.template cast <typename EigenDenseF32::Scalar>();
```

Comment:
We cast to user desired dtype for parsed vertex matrix.

Reply:
perfect